cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Dict, List, Tuple, Union
+from uuid import uuid4
+
+import datumaro as dm
+import django_rq
+from django.conf import settings
+from django.db import transaction
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from cvat.apps.consensus.consensus_reports import (
+    ComparisonReport,
+    generate_assignee_consensus_report,
+    generate_job_consensus_report,
+    generate_task_consensus_report,
+    save_report,
+)
+from cvat.apps.consensus.models import AssigneeConsensusReport, ConsensusReport, ConsensusSettings
+from cvat.apps.consensus.new_intersect_merge import IntersectMerge
+from cvat.apps.dataset_manager.bindings import import_dm_annotations
+from cvat.apps.dataset_manager.task import PatchAction, patch_job_data
+from cvat.apps.engine.models import Job, JobType, StageChoice, StateChoice, Task, User
+from cvat.apps.engine.serializers import RqIdSerializer
+from cvat.apps.engine.utils import (
+    define_dependent_job,
+    get_rq_job_meta,
+    get_rq_lock_by_user,
+    process_failed_job,
+)
+from cvat.apps.quality_control.quality_reports import JobDataProvider
+
+
+class MergeConsensusJobs:
+    def __init__(self, task_id: int) -> None:
+        self.jobs: Dict[int, List[Tuple[int, User]]]
+        self.parent_jobs: List[Job]
+        self.merger: IntersectMerge
+        self.consensus_settings: ConsensusSettings
+        self.assignee_jobs_count: Dict[User, int]
+        self.task_id = task_id
+        self.get_consensus_jobs(task_id)
+        self.get_assignee_jobs_count()
+        self.consensus_settings = ConsensusSettings.objects.filter(task=task_id).first()
+        self.merger = IntersectMerge(
+            conf=IntersectMerge.Conf(
+                pairwise_dist=self.consensus_settings.iou_threshold,
+                output_conf_thresh=self.consensus_settings.agreement_score_threshold,
+                quorum=self.consensus_settings.quorum,
+                sigma=self.consensus_settings.sigma,
+                torso_r=self.consensus_settings.line_thickness,
+            )
+        )
+        if not self.jobs:
+            raise ValidationError(
+                "No annotated consensus jobs found or no regular jobs in annotation stage"
+            )
+
+    def get_consensus_jobs(self, task_id: int) -> None:
```

Comment:
```suggestion
    def _get_consensus_jobs(self, task_id: int) -> None:
```

If this method changes the current object, it's better to make the function private.